VideoLocker/src/main/java/org/edx/mobile/authentication/LoginAPI.java

```diff
@@ -161,38 +166,38 @@ public AuthResponse registerUsingFacebook(@NonNull Bundle parameters, @NonNull S
     }
 
     @NonNull
-    public ResetPasswordResponse resetPassword(@NonNull String email) throws HttpException {
-        return loginService.resetPassword(email);
-    }
-
-    @NonNull
     private void register(Bundle parameters) throws Exception {
```

Comment:
> How difficult would it be to throw a more specific exception?

It shouldn't be too hard; it's only being called by the three type-specific registration methods here, which are in turn being called by `RegisterTask`.

> I know you didnt change the method signature in the effort, but the body did change a lot.

The body was only updated to be compatible with the Retrofit 2 implementation though, by handling error responses manually. There wasn't any essential change to how the exceptions are being thrown. This whole class isn't implemented very well though, as it was a stand-in replacement for the old `Api` class implementation, and the server-side implementation also doesn't seem to be too well defined or implemented. Switching to RxJava will probably make things much simpler here.